File: 0766-toeplitz-matrix/0766-toeplitz-matrix.cpp
class Solution {
public:
    bool isToeplitzMatrix(vector<vector<int>>& matrix) {
        int m = matrix.size() - 1;
        int n = 0;
        for(int i = 0; i < matrix.size() + matrix[0].size() - 1; i++)
        {
            int x = m, y = n;
            while(x < matrix.size() && y < matrix[0].size())
                if(matrix[x++][y++] != matrix[m][n])
                    return false;
            
            if(m > 0)
                m--;
            else
                n++;
        }
        return true;
    }
};

File: 0766-toeplitz-matrix/0766-toeplitz-matrix.py
class Solution:
    def isToeplitzMatrix(self, matrix: List[List[int]]) -> bool:
        m = len(matrix) - 1
        n = 0
        for i in range(len(matrix) + len(matrix[0]) - 1):
            
            x = m
            y = n
            val = matrix[x][y]
            while x < len(matrix) and y < len(matrix[0]):
                if matrix[x][y] != val:
                    return False
                x += 1
                y += 1
            
            if m > 0:
                m -= 1
            else:
                n += 1
        return True